internal/configs/parser_config.go

```diff
@@ -123,9 +123,19 @@ func parseConfigFile(body hcl.Body, diags hcl.Diagnostics, override, allowExperi
 
 				case "state_store":
 					stateStoreCfg, cfgDiags := decodeStateStoreBlock(innerBlock)
```

Comment:
Should the decoding also only happen when experiments are allowed? I believe you can use `innerBlock` for the diagnostics

Reply:
Ah thank you, yes I structured this assuming I needed to output for making the diagnostic - I'll refactor